src/github.com/matrix-org/dendrite/clientapi/auth/storage/devices/devices_table.go

```diff
@@ -140,6 +153,44 @@ func (s *devicesStatements) selectDeviceByToken(
 	return &dev, err
 }
 
+func (s *devicesStatements) selectDeviceByID(
+	ctx context.Context, localpart, device_id string,
```

Comment:
Sorry, but I'm a crank and we use `deviceID` rather than `device_id`... (I've been writing too much python).

This will fix the lints:

>  don't use underscores in Go names; method parameter device_id should be deviceID (golint)